### PR TITLE
[NPU]mindspore model support moe

### PR DESCRIPTION
**Pull request by zzzzzzzxh**
## Motivation
- Enable per‑architecture expert‑location config for MindSpore models, instead of a single generic path.
- Keep dense Qwen3 free from MoE‑specific config while Qwen3 MoE provides it when needed.
- Decouple upstream from downstream details so future MindSpore models can plug in their own behavior cleanly.

## Modifications

- Change MindSporeForCausalLM.get_model_config_for_expert_location to delegate to the selected MindSpore subclass based on HF architectures .
- Implement get_model_config_for_expert_location only in the Qwen3 MoE subclass of sgl_mindspore ; dense Qwen3 does not override (returns None ).
- No changes to the forward/logits path; only the config lookup is updated.
## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [x] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

File: python/sglang/srt/models/mindspore.py [hunk 1]
```diff
@@ -301,5 +302,19 @@ def forward(
         logits_result = LogitsProcessorOutput(next_token_logits=tensor_ms2torch(logits))
         return logits_result
 
+    @classmethod
+    def get_model_config_for_expert_location(cls, config):
```
Comment (chz34) [inline python/sglang/srt/models/mindspore.py:306]: Check and call the corresponding func from self.model,  to support different configuration in sgl_mindspore.  Instead of a general configuration there.

File: python/sglang/srt/models/mindspore.py [hunk 2]
```diff
@@ -28,6 +28,19 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_arch_from_config(config):
+    mindspore_models = import_model_classes("sgl_mindspore.models")
+    architectures = getattr(config, "architectures", [])
+    if isinstance(architectures, str):
+        architectures = [architectures]
+    if not architectures:
+        logger.warning("No model architectures are specified")
```
Comment (ping1jing2) [inline python/sglang/srt/models/mindspore.py]: considering the `ValueError`, is it more appropriate for this branch to raise a error or print warning.

Comment (zzzzzzzxh) [reply python/sglang/srt/models/mindspore.py]: Done. Changed it to raise a ValueError.

**Review by ping1jing2 — APPROVED** [review]
